src/flamenco/vm/fd_vm_private.h

```diff
@@ -184,16 +196,86 @@ fd_vm_mem_cfg( fd_vm_t * vm ) {
    hit loads and pretty good ILP.
 
    fd_vm_mem_haddr_fast is when the vaddr is for use when it is already
-   known that the vaddr region has a valid mapping. */
+   known that the vaddr region has a valid mapping.
+   
+   These assumptions don't hold if direct mapping is enabled since input
+   region lookups become O(log(n)). */
+
+static inline ulong
+fd_vm_get_input_mem_region_idx( fd_vm_t const * vm, ulong offset ) {
```

Comment:
What is the intended behaviour if offset is outside the range of all memory regions? Should this function propagate an error? It would be great to document this, as well as the supported number of regions etc.

Reply:
Should we not throw an error or return a sentinel value if the offset is invalid?

Reply:
I documented that the caller is responsible for handling it. It does so in the relevant syscalls where it gets called and in fd_vm_find_input_mem_region